src/controllers/main/main.ts

```diff
@@ -1207,6 +1207,7 @@ export class MainController extends EventEmitter implements IMainController {
           ? this.networks.networks.filter((n) => chainsToUpdate.includes(n.chainId))
           : undefined
 
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
```

Comment:
I've topped-up the gas tank with the max amount of a token and used the future gas tank value for paying the fee - madness 👏 

But the only thing I noted is that the **pending amount of the token wasn't updated**, even I force reloaded the account few times. After ~1 minute, the token was removed from the dashboard (on pop-up open).